### ISSUE TITLE
examples/basic_firewall

### ISSUE THREAD

**Issue by Pangbit** [dropbox/goebpf#23]
The basic_firewall does not work, I made the following change, can work properly.
Is this related to the environment？

My system environment
kernel: 5.4.2
llvm: 9.0

xdp_fw.c
```
// XDP program //
SEC("xdp")
  ...
  __u64 *rule_idx = bpf_map_lookup_elem(&blacklist, &ip->saddr); 
  ...
}
```

```
// XDP program //
SEC("xdp")
  ...
struct {
    __u32 prefixlen;
    __u32 saddr;
  } key;

  key.prefixlen = 32;
  key.saddr = ip->saddr;

  // Lookup SRC IP in blacklisted IPs
  __u64 *rule_idx = bpf_map_lookup_elem(&blacklist, &key);
  ...
}
```

**Comment by belyalov** [comment]
Thanks for reporting!

You're right, this example is slightly outdated, don't you mind to submit PR with fix?
We would be really appreciated! :)

**Comment by Pangbit** [comment]
I have submitted RP.

**Comment by belyalov** [comment]
Fixed by #24
